sql_generators/urlbar_events/templates/desktop_query.sql

```diff
@@ -1,190 +1,110 @@
+CREATE TEMP FUNCTION enumerated_array(results ARRAY<STRING>, _groups ARRAY<STRING>) AS (
+  ARRAY(
+    SELECT
+      STRUCT(
+        off + 1 AS position,
+        r AS result_type,
+        `mozfun.norm.result_type_to_product_name`(r) AS product_result_type,
+        _groups[SAFE_OFFSET(off)] AS result_group
+      )
+    FROM
+      UNNEST(results) AS r
+      WITH OFFSET off
+  )
+);
+
+CREATE TEMP FUNCTION get_event_action(event_name STRING, engagement_type STRING) AS (
+  CASE
+    WHEN event_name = 'engagement'
+      AND (engagement_type IN ("click", "drop_go", "enter", "go_button", "paste_go"))
+      THEN 'engaged'
+    WHEN event_name = 'abandonment'
+      THEN 'abandoned'
+    WHEN event_name = 'engagement'
+      AND (engagement_type NOT IN ("click", "drop_go", "enter", "go_button", "paste_go"))
+      THEN "annoyance"
```

Comment:
Maybe rearrange these for clarity?

```suggestion
    WHEN event_name = 'engagement'
      AND (engagement_type NOT IN ("click", "drop_go", "enter", "go_button", "paste_go"))
      THEN "annoyance"
    WHEN event_name = 'abandonment'
      THEN 'abandoned'
```